### PR TITLE
Adds a sentient pile of gibs monster for adminbus

### PR DESCRIPTION
**Pull request by yoyobatty**
[Guidelines]: # (Be sure that your PR follows our guidelines, such as modularization and comment standards. You can read more about the subject here: https://github.com/HippieStation/HippieStation/blob/master/hippiestation/README.md )

## Changelog
:cl: YoYoBatty
add: Uh oh, did that wrecked body just move?
/:cl:

## About The Pull Request
This adds a mob that behaves sorta similar to the horroform changeling, but appears to be a pile of sentient gibs, because that's terrifying as fuck. Currently a WIP to figure out a way to cause them to spawn. They will also randomly vent crawl to different parts of the station when it doesn't have any targets. Will also try to devour you which heals them and will continue to attack if you're unconscious.

## Why It's Good For The Game
Spooky hostile mobs are always funny to watch eat half the station.

### CODE REVIEW

**Comment by ghost** [comment]
sounds great,  can you make it gib dead people and dynamically add the gib sprites to its own sprite?

**Comment by ghost** [comment]
gravemind kinda shit

File: hippiestation/code/modules/mob/living/simple_animal/hostile/sentientgibs.dm [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+		adjustBruteLoss(35)
+		visible_message("<span class='warning'>[src] shrivels in reaction to being cleaned!</span>", "<span class='danger'>You can feel your form being disintegrated!.</span>")
+
+/mob/living/simple_animal/hostile/true_changeling/adminbus/gibs/CanAttack(atom/the_target)
+	return TRUE
+
+/mob/living/simple_animal/hostile/true_changeling/adminbus/gibs/AttackingTarget()
+	if(prob(50))
+		if(isliving(target))
+			var/mob/living/lunch = target
```
Comment (Jujumatic) [inline hippiestation/code/modules/mob/living/simple_animal/hostile/sentientgibs.dm]: Holy shit LMAO

Comment (yoyobatty) [reply hippiestation/code/modules/mob/living/simple_animal/hostile/sentientgibs.dm]: It's hungry

**Comment by TothBrush** [comment]
>Currently a WIP to figure out a way to cause them to spawn.
Strange Reagent on gibs maybe? Even funnier, put it in the crafting menu by combining meat and gibs.
or just toss em in the gold slime pool, it would fit right in.

**Comment by Maroonenahmoron** [comment]
I like this idea. Would help convince the crew to clean up their monkey corpses and not throw around their blood tomat- I've said too much.

**Comment by Lxtoofonea** [comment]
+1

**Comment by Manly-Dwarf** [comment]
ok +1

File: hippiestation/code/modules/mob/living/simple_animal/hostile/sentientgibs.dm [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+/mob/living/simple_animal/hostile/true_changeling/adminbus/gibs/Move(atom/newloc, direct)
+	. = ..()
+	if(prob(80))
+		add_splatter_floor(src.loc,FALSE)
+
+/mob/living/simple_animal/hostile/true_changeling/adminbus/gibs/proc/hurt_gibs()
+	if(prob(40))
+		adjustBruteLoss(35)
+		visible_message("<span class='warning'>[src] shrivels in reaction to being cleaned!</span>", "<span class='danger'>You can feel your form being disintegrated!.</span>")
```
Comment (ghost) [inline hippiestation/code/modules/mob/living/simple_animal/hostile/sentientgibs.dm]: `"<span class='danger'>You can feel your form being disintegrated!.</span>"` to "<span class='danger'>You can feel your form being disintegrated!</span>"

**Comment by DaydreamPuppeh** [comment]
+1

File: hippiestation/code/modules/mob/living/simple_animal/hostile/sentientgibs.dm [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+/mob/living/simple_animal/hostile/true_changeling/adminbus/gibs/CanAttack(atom/the_target)
+	if(see_invisible < the_target.invisibility)//Target's invisible to us, forget it
+		return FALSE
+	if(isliving(the_target))
+		var/mob/living/L = the_target
+		if(faction_check_mob(L) && !attack_same)
+			return FALSE
+		return TRUE
+
+/mob/living/simple_animal/hostile/true_changeling/adminbus/gibs/AttackingTarget()
```
Comment (ghost) [inline hippiestation/code/modules/mob/living/simple_animal/hostile/sentientgibs.dm:54]: Not a huge fan of copypasting 60% of horrorform devour code
Could use refactoring that devour itself is an action seperate from selecting a target

**Review by ghost — DISMISSED** [review]

File: hippiestation/code/modules/mob/living/simple_animal/hostile/sentientgibs.dm (hunk 2 above)
Comment (yoyobatty) [reply hippiestation/code/modules/mob/living/simple_animal/hostile/sentientgibs.dm]: yes yes

File: hippiestation/code/modules/mob/living/simple_animal/hostile/sentientgibs.dm (hunk 3 above)
Comment (yoyobatty) [reply hippiestation/code/modules/mob/living/simple_animal/hostile/sentientgibs.dm:54]: I'll improve it in the future, promise.

**Comment by ghost** [comment]
@yoyobatty fuckin finish this

**Pull request merged**